fastseq/optimizer/transformers/modeling_bart_optimizer.py

```diff
@@ -3,247 +3,170 @@
-            if static_kv:
-                new_key_padding_mask = prev_key_padding_mask
-            else:
-                new_key_padding_mask = torch.cat(
-                    [prev_key_padding_mask, key_padding_mask], dim=1)
+            attn_weights_reshaped = None
+
+        attn_probs = nn.functional.dropout(attn_weights, p=self.dropout, training=self.training)
+
+        if is_cross_attention:
+            attn_probs = attn_probs.view(cache_bsz, self.num_beams, self.num_heads, tgt_len, src_len)
+            attn_output = torch.einsum("bmhts,bnhsd->bmhtd", attn_probs, value_states).reshape(-1, tgt_len, self.head_dim)
         else:
-            new_key_padding_mask = key_padding_mask
-        return k, v, new_key_padding_mask
+            attn_output = torch.bmm(attn_probs, value_states)
+        
+        if attn_output.size() != (bsz * self.num_heads, tgt_len, self.head_dim):
+            raise ValueError(
+                f"`attn_output` should be of size {(bsz, self.num_heads, tgt_len, self.head_dim)}, but is {attn_output.size()}"
+            )
 
+        attn_output = attn_output.view(bsz, self.num_heads, tgt_len, self.head_dim)
+        attn_output = attn_output.transpose(1, 2)
+        attn_output = attn_output.reshape(bsz, tgt_len, embed_dim)
 
-@replace(BartForConditionalGeneration)
-class BartForConditionalGenerationV2(BartForConditionalGeneration):
-    """
-    The BART Model with a language modeling head. Can be used for
-    summarization.
-    """
-    @staticmethod
-    def _reorder_cache(past, beam_idx):
-        (encoder_outputs, decoder_past_key_values) = past
-        enc_out, enc_mask = encoder_outputs[:2]
-        reordered_past = decoder_past_key_values
-        if decoder_past_key_values is not None:
-            reordered_past = []
-            for layer_past in decoder_past_key_values:
-                # Get the correct batch idx from decoder layer's batch dim for
-                # self-attn; Note that there is no need to reorder the cached
-                # key and value for the encoder-decoder-attn, because the key
-                # and value for the beams of each sample is the same and we can
-                # cache just one copy to save GPU memory.
-                layer_past_new = {}
-                for attn_key, attn_cache in layer_past.items():
-                    if attn_key == 'self':
-                        layer_past_new[attn_key] = _reorder_buffer(
-                            attn_cache, beam_idx)
-                        continue
-                    layer_past_new[attn_key] = attn_cache
-
-                reordered_past.append(layer_past_new)
-
-        new_enc_mask = (enc_mask if (enc_mask is None or enc_mask == [])
-            else enc_mask.index_select(0, beam_idx))
-
-        past = ((enc_out, new_enc_mask, *encoder_outputs[2:]), reordered_past)
-        return past
-
-MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING[BartConfig] = BartForConditionalGenerationV2 # pylint: disable=line-too-long
```

Comment:
Why is this removed? I think this is used to make `@replace` works as expected when loading Bart model. Maybe the latest hf will not need it, but had better to check if all the replaced functions have been used during the inference.